sorts/dnf_sort.py

```diff
@@ -0,0 +1,77 @@
+"""
+Pure implementation of Dutch national flag problem(dnf) sort algorithm in Python
+Dutch National Flag algorithm is a popular algorithm originally designed by Edsger Dijkstra.
+It is the most optimal sort for 3 unique values (eg. 0,1,2) in an Array.
+dnf sort can sort an array of n size with [0<=a[i]<=2] at guaranteed O(n) complexity in a single pass.
+
+The flag of the Netherlands consists of three colors: white, red, and blue. 
+The task is to randomly arrange balls of white, red, and blue in such a way that balls of the same color are placed together. 
+For DNF (Dutch National Flag), we sort an array of 0, 1, and 2's in linear time that does not consume any extra space. 
+We have to keep in mind that this algorithm can be implemented only on an array that has three unique elements.
+
+1)Time complexity is O(n).
+2)Space complexity is O(1).
+
+More info on: https://en.wikipedia.org/wiki/Dutch_national_flag_problem
+
+For doctests run following command:
+python -m doctest -v dnf_sort.py
+or
+python3 -m doctest -v dnf_sort.py
+
+For manual testing run:
+python dnf_sort.py
+
+"""
+
+
+# Python program to sort an array with
+# 0, 1 and 2 in a single pass
+red = 0 # In this case this is the first color of the flag.
+white = 1 # This is the second color of the flag.
+blue = 2 # This is the third color of the flag.
+colors = [red, white, blue]
+# Function to sort array
+def dnf_sort( sequence:list, arr_size:int)->list:
+    """
+     Pure implementation of dnf sort algorithm in Python
+    :param data: 3 unique values (eg. 0,1,2) in an Array
+    :return: the same collection in ascending order
+    Examples:
+    Input: dnf_sort([2, 1, 0, 0,1, 2],6)
+    Output: [0, 0, 1, 1, 2, 2]
+    Input: dnf_sort([0, 1, 1, 0, 1, 2, 1, 2, 0, 0, 0, 1],12)
+    Output: [0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 2, 2]
+    """
+    if sequence is None: return
+    if len(sequence) <= 1: return
+    low = 0
+    high = arr_size - 1
+    mid = 0
+    while mid <= high:
+        if sequence[mid] == colors[0]:
+            sequence[low], sequence[mid] = sequence[mid], sequence[low]
+            low = low + 1
+            mid = mid + 1
+            continue
+        if sequence[mid] == colors[1]:
+            mid = mid + 1
+            continue
+        if sequence[mid] == colors[2]:
+            sequence[mid], sequence[high] = sequence[high], sequence[mid]
+            high = high - 1
+            continue
+            
+        raise ValueError(f"The elements inside the sequence must contains only {colors} values")    
+    return sequence
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+
+    assert dnf_sort([1, 0, 0, 2, 1],5) == [0,0,1, 1, 2]
+    assert dnf_sort([0, 1, 1, 0, 1, 2, 1, 2, 0, 0, 0, 1],12) == [0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 2, 2]
```

Comment:
```suggestion
```
These can be done as doctests which are both human and machine readable/testaable.